apps/vaporgui/PMetadataSection.cpp

```diff
@@ -0,0 +1,145 @@
+
+namespace {
+    std::vector<std::string> xtypeLookup = { "INVALID", "FLOAT", "DOUBLE", "UINT8", "INT8", "INT32", "INT64", "TEXT" };
+    std::vector<std::string> axisLookup  = { "X", "Y", "Z", "T"};
+}
+
+PMetadataSection::PMetadataSection()
+: PSection("Metadata", { new PMetadataTree() } )
+{
+}
+
+PMetadataTree::PMetadataTree()
+: PWidget("", _tree = new QTreeWidget)
+{
+    _tree->setHeaderHidden(true);
+    _tree->header()->setStretchLastSection(false);
+    _tree->setColumnCount(2);
+    _tree->header()->setSectionResizeMode(QHeaderView::ResizeToContents);
+}
+
+void PMetadataTree::updateGUI() const
+{
+    VAPoR::RenderParams* rp = dynamic_cast<VAPoR::RenderParams*>(getParams());
+    if (rp == nullptr) MSG_ERR("Invalid RenderParams instance for acquiring variable metadata.");
+    VAPoR::DataMgr* dm = getDataMgr();
+    if (rp == nullptr) MSG_ERR("Invalid DataMgr instance for acquiring variable metadata.");
+
+    std::vector<std::string> vars;
+    vars.push_back( rp->GetVariableName() );
+    vars.push_back( rp->GetXFieldVariableName() );
+    vars.push_back( rp->GetYFieldVariableName() );
+    vars.push_back( rp->GetZFieldVariableName() );
+    vars.push_back( rp->GetHeightVariableName() );
+    vars.push_back( rp->GetColorMapVariableName() );
+    
+    std::vector<std::string> auxs = rp->GetAuxVariableNames();
+    vars.insert(vars.end(), auxs.begin(), auxs.end());
+
+    QStringList qvars;
+    for (auto var : vars)
+        qvars << QString::fromStdString(var);
+    qvars.removeDuplicates();
+    qvars.removeAll(QString(""));
+
+    size_t ts = rp->GetCurrentTimestep();
+    int rLevel = rp->GetRefinementLevel();
+    int cLevel = rp->GetCompressionLevel();
+
+    _tree->clear();
+    for (auto qvar : qvars) {
+        std::vector<double> range;
+        dm->GetDataRange(ts, qvar.toStdString(), rLevel, cLevel, range);
+        QTreeWidgetItem* varItem = new QTreeWidgetItem(_tree, {qvar});
+        new QTreeWidgetItem(varItem, {"Min:", QString::number(range[0])});
+        new QTreeWidgetItem(varItem, {"Max:", QString::number(range[1])});
+
+        std::vector<size_t> dims;
+        dm->GetDimLensAtLevel(qvar.toStdString(), rLevel, dims, ts);
+        QString qDims = QString::number(dims[0]) + ":" + QString::number(dims[1]) + ":" + QString::number(dims[2]);
+        new QTreeWidgetItem(varItem, {"Dims (XYZ):", qDims});
+
+        VAPoR::DC::DataVar dataVar;
+        dm->GetDataVarInfo(qvar.toStdString(), dataVar);
+        new QTreeWidgetItem(varItem, {"Units:", QString::fromStdString(dataVar.GetUnits())});
+        _tree->insertTopLevelItem(0,varItem);
+
+        _generateCoordVarInfo(varItem, dataVar);
+        _generateAttributeInfo(varItem, dataVar);
+    }
+}
+
+void PMetadataTree::_generateCoordVarInfo(QTreeWidgetItem* parent, const VAPoR::DC::DataVar dataVar) const {
+    VAPoR::DataMgr* dm = getDataMgr();
+    if (dm == nullptr) MSG_ERR("Invalid DataMgr instance for acquiring coordinate variable metadata.");
```

Comment:
Ditto regarding not proceeding with a null pointer. Probably reasonable to simply VAssert here.

Reply:
Fixed.

Reply:
Not seeing this change either.